src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,40 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final int OPERATION = 0;
+    private static final int AMOUNT = 1;
 
     public void getStatistic(String fromFileName, String toFileName) {
+        int totalSupply = 0;
+        int totalBuy = 0;
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName));
+                BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(toFileName))) {
+            String value = bufferedReader.readLine();
+            while (value != null) {
+                String[] valueArray = value.split(",");
+                String operation = valueArray[OPERATION];
+                int amount = Integer.parseInt(valueArray[AMOUNT]);
+                totalSupply = operation.equals(SUPPLY) ? totalSupply + amount : totalSupply;
+                totalBuy = operation.equals(BUY) ? totalBuy + amount : totalBuy;
+                value = bufferedReader.readLine();
+            }
+            bufferedWriter.write(getResult(totalSupply, totalBuy));
```

Comment:
could we move the logic of writing a report to a separate private method?

Reply:
the logic of creating a report is already placed in a separate method  createReport(). What r u mean?

Reply:
> the logic of creating a report is already placed in a separate method createReport(). What r u mean?

the logic of creating a report is already placed in a separate method

Reply:
Yes, but I ask to move logic of WRITING a file with report to separate private method